core/services/src/main/java/org/exoplatform/services/cms/link/impl/LinkManagerImpl.java

```diff
@@ -444,4 +444,32 @@ public void updateSymlink(Node node) throws RepositoryException {
     }
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Node getLinkInFolder(String contentUUID, String folderPath, String workspace, String nodeType) {
+    ManageableRepository repository = WCMCoreUtils.getRepository();
+    SessionProviderService sessionProviderService = CommonsUtils.getService(SessionProviderService.class);
+    SessionProvider sessionProvider = sessionProviderService.getSystemSessionProvider(null);
+    try {
+      Session session = sessionProvider.getSession(workspace, repository);
+      QueryManager queryManager = session.getWorkspace().getQueryManager();
+      String uuidType = nodeType.equals(NodetypeConstant.EXO_SYMLINK)? NodetypeConstant.EXO_UUID : NodetypeConstant.JCR_UUID;

```

Comment:
This function is just for getting symlinks. You can not add the logic of checking if a node is a child of another node.

Reply:
What about links of document those are not symlinks ? and not saved in the shared folder ?
If we remove this option the sent link in the emails notification of the activity with attached file will be null and empty

Reply:
done